tests/chainer_tests/training_tests/updaters_tests/test_standard_updater.py

```diff
@@ -115,6 +118,283 @@ def test_serialize(self):
         assert serializer.called == [('iteration', 0)]
 
 
+_backend_params = [
+    # NumPy
+    {},
+    # CuPy
+    {'use_cuda': True, 'cuda_device': 0},
+    {'use_cuda': True, 'cuda_device': 1},
+    # ChainerX
+    {'use_chainerx': True, 'chainerx_device': 'native:0'},
+    {'use_chainerx': True, 'chainerx_device': 'cuda:0'},
+    {'use_chainerx': True, 'chainerx_device': 'cuda:1'},
+]
+
+
+@chainer.testing.backend.inject_backend_tests(None, _backend_params)
+@chainer.testing.backend.inject_backend_tests(None, _backend_params)
+@chainer.testing.backend.inject_backend_tests(None, _backend_params)
+class TestStandardUpdaterDevice(unittest.TestCase):
+
+    def test_device(
+            self, model_initial_backend_config, model_backend_config,
+            input_backend_config):
+        model_initial_device = model_initial_backend_config.device
+        device = model_backend_config.device
+        input_device = input_backend_config.device
+
+        model = chainer.Link()
+        model.to_device(model_initial_device)
+        optimizer = DummyOptimizer()
+        optimizer.setup(model)
+        iterator = DummyIterator([numpy.array(1), numpy.array(2)])
+
+        updater = training.updaters.StandardUpdater(
+            iterator,
+            optimizer,
+            device=device,
+            input_device=input_device)
+
+        assert updater.device is device
+        assert updater.input_device is input_device
+
+        # Check the model device.
+        assert model.device == device
+
+        updater.update_core()
+
+        assert optimizer.update.call_count == 1
+        args, kwargs = optimizer.update.call_args
+        assert len(args) == 2
+        assert len(kwargs) == 0
+        loss, v1 = args
```

Comment:
Some minors but how about using a `_` for `loss` as it's intensionally not used and different name for `v1`, .e.g. `in_array`.

Reply:
They're consistent with other tests.